src/integrity-checker.js

```diff
@@ -118,23 +109,59 @@ export default class InstallationIntegrityChecker {
   }
 
   /**
-   * returns a list of files recursively in a directory sorted
+   * Get the list of the directories that contain our modules (there might be multiple such folders b/c of workspaces).
    */
-  async _getFilesDeep(rootDir: string): Promise<Array<string>> {
-    async function getFilePaths(rootDir: string, files: Array<string>, currentDir: string = rootDir): Promise<void> {
-      for (const file of await fs.readdir(currentDir)) {
-        const entry = path.join(currentDir, file);
+
+  _getModulesFolders({workspaceLayout}: {workspaceLayout: ?WorkspaceLayout} = {}): Array<string> {
+    const locations = [];
+
+    if (this.config.modulesFolder) {
+      locations.push(this.config.modulesFolder);
+    } else {
+      locations.push(path.join(this.config.lockfileFolder, 'node_modules'));
+    }
+
+    if (workspaceLayout) {
+      for (const workspaceName of Object.keys(workspaceLayout.workspaces)) {
+        const loc = workspaceLayout.workspaces[workspaceName].loc;
+
+        if (loc) {
+          locations.push(path.join(loc, 'node_modules'));
+        }
+      }
+    }
+
+    return locations;
+  }
+
+  /**
+   * Get a list of the files that are located inside our module folders.
+   */
+  async _getIntegrityListing({workspaceLayout}: {workspaceLayout: ?WorkspaceLayout} = {}): Promise<Array<string>> {
+    const files = [];
+
+    const recurse = async dir => {
+      if (!await fs.exists(dir)) {
```

Comment:
The `dir` might not exists at the root. I'll move it inside the l160 loop, since it only matter there.